### PR TITLE
[TS-1006] Add get_sanitized_object

### PR DESCRIPTION
**Pull request by Phil-Barber**
[TS-1006]
Histories will want access to this same object so figured it made sense
to abstract it slightly.

In histories we will actually also want to filter by dirty fields, but the way
this is used currently I didn't think it made sense to make an abstraction for
it.

- Extract sanitise logic to get_sanitized_object
- Lint touched methods refactored sanise_for_json added doc strings

[TS-1006]: https://lantum.atlassian.net/browse/TS-1006

### CODE REVIEW

File: tests/test_announcechanges.py [hunk 1]
```diff
@@ -309,47 +309,62 @@ def announce_method(message):
...
+    settings.ANNOUNCE_METHOD = get_announce_method(tm)
+    # Save
     with pytest.raises(IntendedException):
         tm.save()
 
+    tm.id = tm.id + 1
+    settings.ANNOUNCE_METHOD = get_announce_method(tm, created=True)
+    # Create
+    with pytest.raises(IntendedException):
+        tm.save(force_insert=True)
```
Comment (Phil-Barber) [inline tests/test_announcechanges.py:366]: I'm not really that familiar with pytest so if there's a better way to do subtest type things let me knowwww

Comment (iponce42) [reply tests/test_announcechanges.py:366]: What do you mean by "subtest type"?

**Comment by lukebryant** [comment]
The ticket says "Update DjangoChangesOnSave to have get_differences method". Are you still working on the get_differences part?

File: tests/test_announcechanges.py (hunk 1 above)
Comment (lukebryant) [reply tests/test_announcechanges.py:366]: @iponce42 it took me a while to work out what phil meant; the diff is obscuring the code 

```   
settings.ANNOUNCE_METHOD = get_announce_method(tm)
with pytest.raises(IntendedException):
    tm.save()	        tm.save()

# Create
tm.id = tm.id + 1

settings.ANNOUNCE_METHOD = get_announce_method(tm, created=True)
  with pytest.raises(IntendedException):
      tm.save(force_insert=True)
```

He's doing the same thing twice but with different arguments for get_announce_method.

if in unittest you could do

```
for args in [{tm}, {tm, 'created':True}]:
    with self.subtest(args=args):
        settings.ANNOUNCE_METHOD = get_announce_method(tm, created=True)
        with self.assertRaises(IntendedException):
            tm.save(force_insert=True)
```

...and they would run kind of as if they were different tests. The first could fail and the second could still pass etc.

@Phil-Barber there's no way I know of doing this in pytest. I don't think this is that great a use-case for subtest anyway. I would just try to abstract out the shared setup...or not bother.

**Comment by Phil-Barber** [comment]
> The ticket says "Update DjangoChangesOnSave to have get_differences method". Are you still working on the get_differences part?

Ah yea so originally we were going to use the exact same differences object, however we realised all we really care about it the after. So instead we are just going to be getting the sanitised object and choosing the fields that changed.

This is what I meant by:
> In histories we will actually also want to filter by dirty fields, but the way
this is used currently I didn't think it made sense to make an abstraction for
it.

We will write some function that looks roughly like:
```
changed_fields = self.get_dirty_fields(check_relationships=True)
sanitised_object = self.get_sanitised_object()
diff = { key: sanitised_object[key] for key in changed_fields }
```

Since this won't be used anywhere inside this module, I figured it doesn't make sense to add here

**Comment by Phil-Barber** [comment]
In an ideal world I would be able to extract this sanitise stuff out completely so that shift histories wouldn't be dependent on changes on save, but its dependent on dirtyFieldsMixin either way so there's no ideal solution I think

**Comment by Phil-Barber** [comment]
Closing in favour of https://github.com/networklocum/session-service/pull/987

**Pull request closed without merging**